k3s_test.go

```diff
@@ -11,6 +12,12 @@ import (
 )
 
 func TestMain(m *testing.M) {
+	// Skip the test if running in CI and not on linux, because we won't be able to run the container other than on linux
+	isCi := os.Getenv("CI")
+	if isCi == "true" && runtime.GOOS != "linux" {
+		return
+	}
+
```

Comment:
1. I'd prefer to call `t.SkipNow()` in the individual tests.
2. Wouldn't it be better to check for `docker` or something? IMO, checking for CI is a bad smell.